ui/lib/sync/addon/components/secrets/page/destinations/destination/sync.ts

```diff
@@ -0,0 +1,166 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { service } from '@ember/service';
+import { action } from '@ember/object';
+import { task } from 'ember-concurrency';
+import { keyIsFolder, parentKeyForKey, keyWithoutParentKey } from 'core/utils/key-utils';
+import errorMessage from 'vault/utils/error-message';
+
+import type SyncDestinationModel from 'vault/models/sync/destination';
+import type KvSecretMetadataModel from 'vault/models/kv/metadata';
+import type RouterService from '@ember/routing/router-service';
+import type StoreService from 'vault/services/store';
+import type FlashMessageService from 'vault/services/flash-messages';
+import type { SearchSelectOption } from 'vault/vault/app-types';
+
+interface Args {
+  destination: SyncDestinationModel;
+}
+interface PowerSelectAPI {
+  actions: {
+    open(): void;
+  };
+}
+
+export default class DestinationSyncPageComponent extends Component<Args> {
+  @service declare readonly router: RouterService;
+  @service declare readonly store: StoreService;
+  @service declare readonly flashMessages: FlashMessageService;
+
+  constructor(owner: unknown, args: Args) {
+    super(owner, args);
+    this.fetchMounts();
+  }
+
+  @tracked mounts: SearchSelectOption[] = [];
+  @tracked secrets: KvSecretMetadataModel[] = [];
+  @tracked mountPath = '';
+  @tracked secretPath = '';
+  @tracked error = '';
+  powerSelectAPI: PowerSelectAPI | undefined;
+  _lastFetch: KvSecretMetadataModel[] | undefined; // cache the response for filtering purposes
+
+  // strip trailing slash from mount path
+  get mountName() {
+    return keyIsFolder(this.mountPath) ? this.mountPath.slice(0, -1) : this.mountPath;
+  }
+
+  // unable to use built-in fetch functionality of SearchSelect since we need to filter by kv type
+  async fetchMounts() {
+    try {
+      const secretEngines = await this.store.query('secret-engine', {});
+      this.mounts = secretEngines.reduce((filtered, model) => {
+        if (model.type === 'kv') {
```

Comment:
Is it any KV backend, or only version 2? Also, if it's only version 2 we should probably note that type `generic` can be upgraded to version 2 and I think is expected to have the same behavior as kv v2?

Reply:
That's a good question. I'll double check on this and update accordingly if need be.

Reply:
Confirmed that it's only kv v2 at the moment. I updated the filter function. Thanks for bringing up the question!